### PR TITLE
fix typos of controller mapping dialog and virtual SB dialog.

### PR DESCRIPTION
**Pull request by jackchentwkh**
fix typos of controller mapping dialog and virtual SB dialog.
correct control names accordingly.

but cxbx.rc still got errors with VS compiler and resource editor.
need further investigation.

### CODE REVIEW

**Comment by RadWolfie** [comment]
This pull request is corrupted. Please start the clone on top of latest master.

I did not revert `.gitattributes` btw. @jagotu, please do not alternate the file. The issue came from this pull request https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/pull/1218

**Comment by jackchentwkh** [comment]
the USB pass through for SB controller is sort of working. there is a obvious input lag which I shall investigate later. I need suggestions. I am suspecting the stream read/write code is the root cause because the stream read will cache the input sate changes in a queue.
As for the commits, I rebase to latest master and the rebase to my remote repo before I push the update. I can't push to my remote repo witout rebase to it.
I don't know why those submodule changes are included in the commits.
how shall I fix these unwanted commits?

**Comment by jackchentwkh** [comment]
ok, fixed the input/feedback lag finally. it all seems working good.
now can someone help me how to make this PR into a more clear state for being merged?

**Comment by LukeUsher** [comment]
@jackchentwkh you need to have git setup with two remotes: One remote should be 'origin' and is your fork, the other remote should be 'upstream' and that should be the Cxbx-Reloaded master.

Then on the command-line, you can simply do the following to rebase:
```sh
git checkout <branchname>  #Switch to the desired branch
git fetch upstream         # Fetch upstream changes
git rebase upstream/master # Rebase the current branch on upstream/master
git push -f                # Force a push (force required because rebasing rewrites history)
```

**Comment by jackchentwkh** [comment]
@LukeUsher thank you, I did what you said, rebase to upstream/master, and push -f, it worked. 
but there is one submodule file d3d8to9.cpp marked as changed, I am not sure how this happen.
git shows no actual difference between my branch and master.

File: build/win32/CxbxVSBC.vcxproj [hunk 1]
```diff
@@ -85,19 +85,22 @@
...
     <ClCompile>
       <WarningLevel>Level3</WarningLevel>
-      <Optimization>Full</Optimization>
+      <Optimization>Disabled</Optimization>
       <FunctionLevelLinking>true</FunctionLevelLinking>
       <IntrinsicFunctions>true</IntrinsicFunctions>
       <SDLCheck>true</SDLCheck>
       <PreprocessorDefinitions>WIN32;NDEBUG;CXBXVSBC_EXPORTS;_WINDOWS;_USRDLL;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <ConformanceMode>true</ConformanceMode>
+      <AdditionalIncludeDirectories>C:\Users\jack\Source\Repos\Cxbx-Reloaded\import\libusbK\includes;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (PatrickvL) [inline build/win32/CxbxVSBC.vcxproj]: Change this into a relative path

File: build/win32/CxbxVSBC.vcxproj [hunk 2]
```diff
@@ -85,19 +85,22 @@
...
     </ClCompile>
     <Link>
       <SubSystem>Windows</SubSystem>
       <EnableCOMDATFolding>true</EnableCOMDATFolding>
       <OptimizeReferences>true</OptimizeReferences>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <ModuleDefinitionFile>CxbxVSBC.def</ModuleDefinitionFile>
+      <AdditionalLibraryDirectories>C:\Users\jack\Source\Repos\Cxbx-Reloaded\import\libusbK\lib\x86;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
```
Comment (PatrickvL) [inline build/win32/CxbxVSBC.vcxproj]: Also relative path here

File: build/win32/CxbxVSBC.vcxproj [hunk 3]
```diff
@@ -107,11 +110,14 @@
       <SDLCheck>true</SDLCheck>
       <PreprocessorDefinitions>WIN32;_DEBUG;CXBXVSBC_EXPORTS;_WINDOWS;_USRDLL;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <ConformanceMode>true</ConformanceMode>
+      <AdditionalIncludeDirectories>C:\Users\jack\Source\Repos\Cxbx-Reloaded\import\libusbK\includes;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (PatrickvL) [inline build/win32/CxbxVSBC.vcxproj]: And here

File: build/win32/CxbxVSBC.vcxproj [hunk 4]
```diff
@@ -107,11 +110,14 @@
...
     </ClCompile>
     <Link>
       <SubSystem>Windows</SubSystem>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <ModuleDefinitionFile>CxbxVSBC.def</ModuleDefinitionFile>
+      <AdditionalLibraryDirectories>C:\Users\jack\Source\Repos\Cxbx-Reloaded\import\libusbK\lib\x86;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
```
Comment (PatrickvL) [inline build/win32/CxbxVSBC.vcxproj]: And here

File: import/d3d8to9 [hunk 5]
```diff
@@ -1 +1 @@
-Subproject commit c7236282d5c3f8ee871e26953cfdd2eeebe964cf
+Subproject commit 0b199c25feabc9c0f5f9a5005106f5bc8b935005
```
Comment (PatrickvL) [inline import/d3d8to9]: Better not do such an unrelated change

Comment (jackchentwkh) [reply import/d3d8to9]: I didn't do that :(

File: build/win32/CxbxVSBC.vcxproj (hunk 1 above)
Comment (jackchentwkh) [reply build/win32/CxbxVSBC.vcxproj]: all 4 path are fixed

File: build/win32/CxbxVSBC.vcxproj (hunk 2 above)
Comment (jackchentwkh) [reply build/win32/CxbxVSBC.vcxproj]: all 4 path are fixed

File: build/win32/CxbxVSBC.vcxproj (hunk 3 above)
Comment (jackchentwkh) [reply build/win32/CxbxVSBC.vcxproj]: all 4 path are fixed

File: build/win32/CxbxVSBC.vcxproj (hunk 4 above)
Comment (jackchentwkh) [reply build/win32/CxbxVSBC.vcxproj]: all 4 path are fixed

**Comment by LukeUsher** [comment]
A PR should ideally contain one single unit of work: THis PR should have been made as at least two, maybe 3 different PRs.

1. Fix typos of controller mapping dialog and virtual SB dialog. 
2. restore cxbx.rc encoding to unicode (utf16)
3. USB Steelbattaleon passthrough support

This would allow discussion/testing on each individual change, and the first two could have been merged already.

The 3rd task requires more testing/discussion because it brings in a new 3rd party library (libusbk), we need to make sure that 1. The licenses are compatible and 2. It is the best available library for USB pass-through. (It'd suck if we needed a second dependency for LLE USB passthrough, when it eventually comes!)

**Comment by LukeUsher** [comment]
Additionally, why is libusbk copied in, and not imported into a submodule?
Using a submodule makes it much easier to keep up-to-date with upstream changes, and keeps the Cxbx-R repository clean.

If we needed to make modifications to the library to work for us, then we should fork it into the Cxbx-Reloaded organization and import that fork as a submodule.

**Comment by jackchentwkh** [comment]
Luke, I agree with you regarding to the lib library should be included with source as submodule. But frankly that’s beyond my current skill set. The reason why I choose Libusb is that there is a windows utility available to access SB controller which uses libusb library. The decision is up to the team.

**Comment by LukeUsher** [comment]
Feature-Freeze is happening now. I'll glady accept the first commit of this pr (2ad3f2bad6644e56925a29ce6a078a4ff9aba660) if you can create one that applies only that change.

The rest of this PR will have to wait, as USB pass-through is quite a large feature.

**Comment by PatrickvL** [comment]
@ergo720 told me in so many words, that libusbk might not be a wise choice - a better choice (which has the benefit of being cross-platform) is : [libusb](https://github.com/libusb/libusb/wiki/FAQ#libusborg_libusbxorg_and_libusbinfo)

It also supports Xbox controllers out of the box, see https://github.com/libusb/libusb/blob/6402a10c145d1a38964fc43009d115fe7a86c8c7/examples/xusb.c#L282 and https://github.com/libusb/libusb/blob/6402a10c145d1a38964fc43009d115fe7a86c8c7/examples/xusb.c#L950

**Comment by jackchentwkh** [comment]
Well, I have no objections at all. It’s not in my knowledge base. Let the team discuss and come out the conclusion and I will revise the code accordingly. Libusbk is part of libusb as I read the faq. It shouldn’t be too complex to revised the code any way.

**Comment by PatrickvL** [comment]
Hi Jack, let's have a team discussion on this on discord. Otherwise this keeps dragging on for no good reason. Plus, we don't want to keep you waiting either.

**Comment by jackchentwkh** [comment]
Ok

**Comment by jackchentwkh** [comment]
@PatrickvL ergo is writing code for LLE USB. I think we should wait for his PR and see what's there first. 
after that, we could start our discussion.

**Comment by PatrickvL** [comment]
Fair points. Perhaps it's best for now to close this pull request without merging?

**Comment by jackchentwkh** [comment]
agree
